polaris/hub/client.py

```diff
@@ -153,22 +154,28 @@ def _load_from_signed_url(self, url: URLTypes, load_fn: Callable, headers: Heade
         content = BytesIO(response.content)
         return load_fn(content)
 
-    def _base_request_to_hub(self, url: str, method: str, **kwargs):
+    def _base_request_to_hub(self, url: str, method: str, show_informative_error: bool = False, **kwargs):
         """Utility function since most API methods follow the same pattern"""
         response = self.request(url=url, method=method, **kwargs)
 
         try:
             response.raise_for_status()
 
         except HTTPStatusError as error:
+            response_status_code = response.status_code
+
             # With an internal server error, we are not sure the custom error-handling code on the hub is reached.
-            if response.status_code == 500:
+            if response_status_code == 500:
                 raise
 
             # If not an internal server error, the hub should always return a JSON response
             # with additional information about the error.
             response = response.json()
             response = json.dumps(response, indent=2, sort_keys=True)
+
+            if show_informative_error and (response_status_code == 403 or response_status_code == 404):
```

Comment:
I'm not sure if we want the `response_status_code == 404` here. I added it because users facing this data discrepancy issue could run into 404s despite them having the correct permissions. 

So in that case, it seems suitable here, but I wanted to gather some additional opinions on it.

Reply:
When does the user get a `404` when they shouldn't? When the user has gained a membership in Clerk that isn't reflected in the JWT they'd get a `403`, right?

Reply:
> When the user has gained a membership in Clerk that isn't reflected in the JWT they'd get a 403, right?

When retrieving datasets you don't have access to, you would get a 404. So if you just joined an organization and try download a private dataset, that could result in a 404 when the JWT is outdated.